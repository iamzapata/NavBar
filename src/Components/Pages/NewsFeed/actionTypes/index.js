import keyMirror from "keymirror"

const ActionTypes = keyMirror({
  GET_NEWS_FEED_REQUEST: null,
  GET_NEWS_FEED_SUCCESS: null,
  GET_NEWS_FEED_FAILURE: null
})

export default ActionTypes
