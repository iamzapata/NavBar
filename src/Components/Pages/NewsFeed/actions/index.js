import ActionTypes from "../actionTypes"
import request from "utils/request"

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms))
}

const getNewsFeedRequest = () => {
  return {
    type: ActionTypes.GET_NEWS_FEED_REQUEST,
    isLoading: true
  }
}

const getNewsFeedSuccess = feed => {
  return {
    type: ActionTypes.GET_NEWS_FEED_SUCCESS,
    isLoading: false,
    feed
  }
}

const getNewsFeedFailure = err => {
  return {
    type: ActionTypes.GET_NEWS_FEED_FAILURE,
    isLoading: false,
    err
  }
}

export default function getNewsFeed() {
  return async dispatch => {
    dispatch(getNewsFeedRequest())
    try {
      await sleep(500)
      const feed = await request("feed/index.json")
      dispatch(getNewsFeedSuccess(feed))
    } catch (err) {
      dispatch(getNewsFeedFailure(err))
    }
  }
}
