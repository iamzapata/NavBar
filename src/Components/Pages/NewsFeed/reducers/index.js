import { createSelector } from "reselect"
import ActionTypes from "../actionTypes"

const defaultState = {
  feed: [],
  isLoading: false,
  err: null
}

export default function NewsFeedStore(state = defaultState, action) {
  const { type, feed, isLoading, err } = action
  switch (type) {
    case ActionTypes.GET_NEWS_FEED_REQUEST:
      return {
        ...state,
        isLoading
      }
    case ActionTypes.GET_NEWS_FEED_SUCCESS:
      return {
        ...state,
        feed,
        isLoading
      }
    case ActionTypes.GET_NEWS_FEED_FAILURE:
      return {
        ...state,
        isLoading,
        err
      }
    default:
      return state
  }
}

const getNewsFeed = state => state.NewsFeed.feed

export const newsFeedSelector = createSelector([getNewsFeed], feed => feed)
