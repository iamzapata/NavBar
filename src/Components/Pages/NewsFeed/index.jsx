import React, { PureComponent } from "react"
import { arrayOf, func, shape, bool } from "prop-types"
import { connect } from "react-redux"
import Spinner from "Components/Common/Spinner"
import PageContainer from "Pages/PageContainer"
import getNewsFeed from "./actions"
import { newsFeedSelector, newsFeedLoadingSelector } from "./reducers"
import FeedList from "./Components/FeedList"

class NewsFeed extends PureComponent {
  componentDidMount() {
    const { getNewsFeed } = this.props
    getNewsFeed()
  }
  render() {
    const { feed, feedLoading } = this.props
    return (
      <PageContainer>
        <div className="columns">
          <div className="column" />
          <div className="column is-3" />
          <FeedList feed={feed} />
          <div className="column" />
        </div>
        <Spinner isLoading={feedLoading} />
      </PageContainer>
    )
  }
}

const mapStateToProps = state => ({
  feed: newsFeedSelector(state),
  feedLoading: newsFeedLoadingSelector(state)
})

const mapDispatchToProps = {
  getNewsFeed
}

NewsFeed.propTypes = {
  feed: arrayOf(shape({})).isRequired,
  feedLoading: bool.isRequired,
  getNewsFeed: func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewsFeed)
