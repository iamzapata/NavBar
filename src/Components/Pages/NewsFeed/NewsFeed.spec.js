import React from "react"
import { render, cleanup } from "react-testing-library"
import configureMockStore from "redux-mock-store"
import { Provider } from "react-redux"
import ReduxThunk from "redux-thunk"
import NewsFeed from "./index"

const mockStore = configureMockStore([ReduxThunk])
const store = mockStore({
  Theme: { themeName: "Pinterest" },
  NewsFeed: { feed: [], isLoading: false },
  UserProfile: { profile: {}, isLoading: false }
})

describe("<NewsFeed />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(
      <Provider store={store}>
        <NewsFeed />
      </Provider>
    )
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
