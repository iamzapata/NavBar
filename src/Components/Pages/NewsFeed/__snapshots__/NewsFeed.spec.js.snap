// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsFeed /> Should render self 1`] = `
<div
  class="PageContainer container"
>
  <div
    class="columns"
  >
    <div
      class="column"
    />
    <div
      class="column is-3"
    />
    <div
      class="FeedList column is-7"
    >
      <span
        class="FeedList__Header"
      >
        LATEST
      </span>
      <div
        class="FeedList__Items"
      />
    </div>
    <div
      class="column"
    />
  </div>
  <div
    class="sk-fading-circle"
    style="width: 50px; height: 50px; display: none;"
  >
    <div
      class="sk-circle1 sk-circle"
    />
    <div
      class="sk-circle2 sk-circle"
    />
    <div
      class="sk-circle3 sk-circle"
    />
    <div
      class="sk-circle4 sk-circle"
    />
    <div
      class="sk-circle5 sk-circle"
    />
    <div
      class="sk-circle6 sk-circle"
    />
    <div
      class="sk-circle7 sk-circle"
    />
    <div
      class="sk-circle8 sk-circle"
    />
    <div
      class="sk-circle9 sk-circle"
    />
    <div
      class="sk-circle10 sk-circle"
    />
    <div
      class="sk-circle11 sk-circle"
    />
    <div
      class="sk-circle12 sk-circle"
    />
  </div>
</div>
`;
